warehouse/csp.py

```diff
@@ -75,6 +78,73 @@ def csp_factory(_, request):
         return CSPPolicy({})
 
 
+def _connect_src_settings(config) -> list:
+    settings = [
+        SELF,
+        "https://api.github.com/repos/",
+        "https://api.github.com/search/issues",
+        "https://*.google-analytics.com",
+        "https://*.analytics.google.com",
+        "https://*.googletagmanager.com",
+        "fastly-insights.com",
+        "*.fastly-insights.com",
+        "*.ethicalads.io",
+        "https://api.pwnedpasswords.com",
+        # Scoped deeply to prevent other scripts calling other CDN resources
+        "https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/sre/mathmaps/",
+    ]
+
+    settings.extend(
+        [item for item in [config.registry.settings.get("statuspage.url")] if item]
+    )
+
+    if config.registry.settings.get("warehouse.env") == Environment.development:
+        livereload_url = config.registry.settings.get("livereload.url")
+        parsed_url = urllib.parse.urlparse(livereload_url)
+
+        # Incoming scheme could be http or https.
+        scheme_replacement = "wss" if parsed_url.scheme == "https" else "ws"
+
+        replaced = parsed_url._replace(scheme=scheme_replacement)  # noqa
+        fixed = urllib.parse.urlunparse(replaced)
+
+        settings.extend(
+            [
+                f"{fixed}/livereload",
+            ]
+        )
```

Comment:
After sleeping on this, this could be simplified to use a wildcard scheme, and since it's development.